AutoMapper.Data/Configuration/Conventions/DataRecordMemberConfiguration.cs

```diff
@@ -147,5 +150,36 @@ private bool IsNestedObject(Type type)
 
             return !resolvedtype.IsPrimitive();
         }
+
+        private string TransformNameFromSourceNamingConvention(TypeDetails sourceTypeDetails, string nameToSearch)
+        {
+            // Sadly, Automapper's INamingConvention doesn't implement method to transform the nameToSearch.
+            // Added support for LowerUnderscoreNamingConvention only.
+            
+            var memberConfiguration = sourceTypeDetails.Config.MemberConfiguration;
+
+            var sourceNamingConvention = memberConfiguration.SourceNamingConvention;
+
+            var destinationNamingConvention = memberConfiguration.DestinationNamingConvention;
+            
+            if (sourceNamingConvention is LowerUnderscoreNamingConvention)
```

Comment:
This is _completely_ ad hoc and has no place in a reusable library. But you can do that in your own fork.